tests/difftests/tests/Cargo.toml

```diff
@@ -44,15 +45,18 @@ unexpected_cfgs = { level = "allow", check-cfg = [
     'cfg(target_arch, values("spirv"))'
 ] }
 
+# Cargo Windows bug: workspace dependencies of crates.io dependencies are resolved
+# incorrectly in the root workspace instead of this difftest workspace. So all
+# workspace dependencies here must be mirrored into the root workspace as well.
```

Comment:
I still don't fully understand what's causing this, I tried creating a minimal repro repo but haven't been able to. how difftest calls cargo (`--manifest-path` or inherited env vars) seems to be important in the repo, as you can just `cargo build` in the `tests/difftest/tests` repo and it'll build just fine.